platform/iphone/detect.py

```diff
@@ -108,28 +101,15 @@ def configure(env):
         detect_darwin_sdk_path("iphone", env)
         env.Append(CCFLAGS=["-miphoneos-version-min=11.0"])
 
-    if env["arch"] == "x86" or env["arch"] == "x86_64":
+    if env["arch"] == "x86_64":
         env["ENV"]["MACOSX_DEPLOYMENT_TARGET"] = "10.9"
-        arch_flag = "i386" if env["arch"] == "x86" else env["arch"]
         env.Append(
             CCFLAGS=(
-                "-fobjc-arc -arch "
-                + arch_flag
-                + " -fobjc-abi-version=2 -fobjc-legacy-dispatch -fmessage-length=0 -fpascal-strings -fblocks"
+                "-fobjc-arc -arch x86_64"
+                " -fobjc-abi-version=2 -fobjc-legacy-dispatch -fmessage-length=0 -fpascal-strings -fblocks"
```

Comment:
Now that there's only two arches, I guess there would be an option to split the shared CCFLAGS outside the `if` check, and then just do arch-based appends for the flags which are unique for each platform (and ideally we should document why we use those flags for each). (Not necessarily in this PR.)